transactions/WFLY-17742_SupportTransactionsDuringGracefulShutdown

```diff
@@ -0,0 +1,155 @@
+* Integrating with the existing logging mechanism of WildFly, users should be notified when subsystems are delaying WildFly’s suspension. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to a user in an understandable manner without exposing internal technical implementation details.
+  * ~~Notifications from the Transactions subsystem should explicitly report enough information to identify the transactions that are delaying the graceful shutdown of WildFly. Moreover, notifications should report whether the transaction is in-doubt or heuristic.~~ This is already implemented in the Transactions subsystem, i.e. details about transactions that need to be recovered (automatically or manually) are reported using the existing logging mechanisms of WildFly.
+
+=== Nice-to-Have Requirements
+
+* Introducing an asynchronous graceful shutdown would enable all ServerActivity implementations to receive a pre-suspend/suspend signal concurrently. In this way, the timeout will have the same duration for all ServerActivity implementations.
+* [The sequence to suspend](https://github.com/wildfly/wildfly-core/blob/main/server/src/main/java/org/jboss/as/server/suspend/SuspendController.java#L57-L99) ServerActivity implementations during WildFly’s graceful shutdown should be Last Input First Output (LIFO), i.e. the last ServerActivity implementation that was registered during startup should be the first one to get suspended.
+
+=== Non Requirements
+
+* As discussed previously, in a cloud environment, when the hosting entity crashes, its state should be recovered by the cloud environment without the intervention of the user
+
+== Implementation Plan
+
+*Aim of the proposal*. All subsystems that have the ability to start transactions should modify their suspension handlers to properly delay WildFly’s graceful shutdown as long as there are transactions to complete. Of course, users should be notified when subsystems are delaying WildFly’s suspension. This is especially true when a negative timeout is used (i.e. indefinite graceful shutdown). Moreover, internal details like 'ServerActivity' do not need to be prominent. Part of working out such a notification would be how to consistently describe a ServerActivity to a user in an understandable manner without exposing internal technical implementation details.
```

Comment:
This proposal was mostly concerned about the Transactions subsystem and the EJB remoting subsystem. Having said that, I should double check if RTS, LRA and XTS need attention (even though I doubt that they will as they are all part of Narayana thus their transactiond are already stored in the object store/log store)

Reply:
IIUC, the list is not definitive and may change in the future, so lets keep this open. I also think Undertow should be there since it you can start a transaction within a servlet method. Same for JAX-RS resources (I am not sure whether Undertow handles these or some other subsystem does).

Reply:
Yeah, let's keep this open. Thanks Ivan.